File: Murz1k.DrinksVendingMachine/Controllers/AdminController.cs
using Murz1k.DrinksVendingMachine.Models;
using Murz1k.DrinksVendingMachine.Services;
using Murz1k.DrinksVendingMachine.Utilities;
using System;
using System.IO;
using System.Web.Mvc;

namespace Murz1k.DrinksVendingMachine.Controllers
{
    [AdminAccess]
    public class AdminController : Controller
    {
        private BeverageService _service;
        private CoinService _coinService;

        public AdminController()
        {
            _service = new BeverageService();
            _coinService = new CoinService();
        }
        public ActionResult Index(string secretKey="")
        {
            if(secretKey=="6396482g35ghlh5gl384")
            {
                IoCProvider.TimeSession = DateTime.Now + new TimeSpan(0, 5, 0);
                return View();
            }
            return RedirectToAction("Index", "Home");
        }

        public void Add(Beverage beverage)
        {
            _service.AddBeverage(beverage);
        }

        public void Edit(int id, Beverage beverage)
        {
            _service.EditBeverage(id, beverage);
        }

        public void Delete(int id)
        {
            var path = _service.GetBeverage(id).ImagePath;
            DeleteImage(path);
            _service.DeleteBeverage(id);
        }

        public void AddCoin(Coin coin)
        {
            _coinService.AddCoin(coin);
        }

        public void EditCoin(int id, Coin coin)
        {
            _coinService.EditCoin(id, coin);
        }
        
        public void DeleteCoin(int id)
        {
            _coinService.DeleteCoin(id);
        }

        private void DeleteImage(string path)
        {
            System.IO.File.Delete(Server.MapPath("~"+path));
        }
        
        public void SaveImage()
        {
            var file = Request.Files[0];
            var stream = file.InputStream;
            var fileName = Path.GetFileName(file.FileName);
            var path = Path.Combine(Server.MapPath("~/Content/Images"), fileName);
            using (var fileStream = System.IO.File.Create(path))
            {
                stream.CopyTo(fileStream);
            }
        }
    }
}

File: Murz1k.DrinksVendingMachine/Controllers/AdminController.js
var app = angular.module("app", []);

app.controller("AdminController", function ($scope) {
    $scope.load = function () {
        $scope.Refresh();
    };
    $scope.path = "/Content/Images/";
    $scope.loadImage = function (input) {
        $scope.$applyAsync(function () {
            var photofile = input.files[0];
            var reader = new FileReader();
            reader.onload = function (e) {
                document.getElementById("image").src = e.target.result;
                var data = new FormData();
                data.append("file", photofile);
                $scope.file = data;
                $scope.filePath = $scope.path + photofile.name;
            };
            reader.readAsDataURL(photofile);
        });
    };
    $scope.dragOver = function (input) {
        evt.stopPropagation();
        evt.preventDefault();
        evt.dataTransfer.dropEffect = 'copy';
    };
    $scope.fileSelect = function (input) {
        $scope.$applyAsync(function () {
            input.stopPropagation();
            input.preventDefault();

            var files = input.dataTransfer.files;
            var photofile = files[0];
            var reader = new FileReader();
            reader.onload = function (e) {
                document.getElementById("image").src = e.target.result;
                var data = new FormData();
                data.append("file", photofile);
                $scope.file = data;
                $scope.filePath = $scope.path + photofile.name;
            };
            reader.readAsDataURL(photofile);

        });
    };
    $scope.SaveImage = function () {
        $scope.currentBeverage.ImagePath = $scope.filePath;
        $.ajax({
            type: "POST",
            url: '/Admin/SaveImage',
            contentType: false,
            processData: false,
            data: $scope.file
        });
    };
    $scope.Refresh = function () {
        $scope.clear();
        $.ajax({
            url: "/Home/GetAll",
            type: "post",
            dataType: "json",
            contentType: "application/json",
            success: function (beverages) {
                $scope.beverages = beverages;
                $.ajax({
                    url: "/Admin/GetAllCoins",
                    type: "post",
                    dataType: "json",
                    contentType: "application/json",
                    success: function (coins) {
                        $scope.coins = coins;
                        $scope.$applyAsync();
                    }
                });
            }
        });
    };
    $scope.currentBeverage = {
        Id: -1,
        Title: "",
        Count: -1,
        Price: -1,
        ImagePath: ""
    };
    $scope.currentCoin = {
        Id: -1,
        Value: 0,
        Count: -1,
        IsLocked: true
    };
    $scope.addCoin = function () {
        if (confirm("Вы точно хотите добавить монету?"))
            $.ajax({
                url: "/Admin/AddCoin",
                data: { coin: $scope.currentCoin },
                type: "post",
                dataType: "html",
                success: function () {
                    $scope.Refresh();
                }
            });
    };
    $scope.editCoin = function () {
        if ($scope.currentCoin.Id < 0) {
            alert("Выберите монету.");
            return 0;
        }
        if (confirm("Вы точно хотите изменить монету?"))
            $.ajax({
                url: "/Admin/EditCoin",
                data: { id: $scope.currentCoin.Id, coin: $scope.currentCoin },
                type: "post",
                dataType: "html",
                success: function () {
                    $scope.Refresh();
                }
            });
    };
    $scope.deleteCoin = function () {
        if ($scope.currentCoin.Id < 0) {
            alert("Выберите монету.");
            return 0;
        }
        if (confirm("Вы точно хотите удалить монету?"))
            $.ajax({
                url: "/Admin/DeleteCoin",
                data: { id: $scope.currentCoin.Id },
                type: "post",
                dataType: "html",
                success: function () {
                    $scope.Refresh();
                }
            });
    };
    $scope.addBeverage = function () {
        if (confirm("Вы точно хотите добавить напиток?"))
            $scope.SaveImage();
            $.ajax({
                url: "/Admin/Add",
                data: { beverage: $scope.currentBeverage },
                type: "post",
                dataType: "html",
                success: function () {
                    $scope.Refresh();
                }
            });
    };
    $scope.editBeverage = function () {
        if ($scope.currentBeverage.Id < 0) {
            alert("Выберите напиток.");
            return 0;
        }
        if (confirm("Вы точно хотите изменить напиток?"))
            $.ajax({
                url: "/Admin/Edit",
                data: { id: $scope.currentBeverage.Id, beverage: $scope.currentBeverage },
                type: "post",
                dataType: "html",
                success: function () {
                    $scope.Refresh();
                }
            });
    };
    $scope.deleteBeverage = function () {
        if ($scope.currentBeverage.Id < 0) {
            alert("Выберите напиток.");
            return 0;
        }
        if (confirm("Вы точно хотите удалить напиток?"))
            $.ajax({
                url: "/Admin/Delete",
                data: { id: $scope.currentBeverage.Id },
                type: "post",
                dataType: "html",
                success: function () {
                    $scope.Refresh();
                }
            });
    };
    $scope.clear = function () {
        $scope.currentBeverage = {
            Id: -1,
            Title: "",
            Count: -1,
            Price: -1,
            ImagePath: ""
        };
    };
    $scope.clearCoin = function () {
        $scope.currentCoin = {
            Id: -1,
            Value: -1,
            Count: -1,
            IsLocked: true
        };
    };
    $scope.chooseBeverage = function (beverage) {
        $scope.currentBeverage = {
            Id: beverage.Id,
            Title: beverage.Title,
            Count: beverage.Count,
            Price: beverage.Price,
            ImagePath: beverage.ImagePath
        };
    };
    $scope.chooseCoin = function (coin) {
        $scope.currentCoin = {
            Id: coin.Id,
            Count: coin.Count,
            IsLocked: coin.IsLocked,
            Value: coin.Value
        };
    };
});